aspnetcore/blazor/forms-validation.md

```diff
@@ -368,6 +366,31 @@ Assign a custom template to <xref:Microsoft.AspNetCore.Components.Forms.InputDat
 
 ::: moniker-end
 
+## Basic validation
+
+In basic form validation scenarios, an <xref:Microsoft.AspNetCore.Components.Forms.EditForm> instance can use declared <xref:Microsoft.AspNetCore.Components.Forms.EditContext> and <xref:Microsoft.AspNetCore.Components.Forms.ValidationMessageStore> instances to validate form fields. A handler for the <xref:Microsoft.AspNetCore.Components.Forms.EditContext.OnValidationRequested> event of the <xref:Microsoft.AspNetCore.Components.Forms.EditContext> executes custom validation logic. The handler's result updates the <xref:Microsoft.AspNetCore.Components.Forms.ValidationMessageStore> instance.
+
+> [!WARNING]
+> When used the same component, the approach in this section is incompatible with the use of a *validator component*, which is described in the [Validator components](#validator-components) section.
+>
+> Use of both approaches simultaneously leads to a conflict over control of the form's <xref:Microsoft.AspNetCore.Components.Forms.ValidationMessageStore>. Basic form validation is most useful in cases where the form's model is defined within the component hosting the form, either as members directly on the component or in a subclass. Use of a validator component is a better choice where an independent model class is used across several components.
+>
+> The approach demonstrated in this section can be used as a shared component in other components that use an <xref:Microsoft.AspNetCore.Components.Forms.EditForm> with validator components. The shared component provides one or more values to its parent component via [component parameters](xref:blazor/components/index#component-parameters).
+
+`Pages/FormExample4.razor`:
+
+::: moniker range=">= aspnetcore-5.0"
+
+[!code-razor[](~/blazor/common/samples/5.x/BlazorSample_WebAssembly/Pages/forms-and-validation/FormExample4.razor?highlight=38,42-53,71)]
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-5.0"
+
+[!code-razor[](~/blazor/common/samples/3.x/BlazorSample_WebAssembly/Pages/forms-and-validation/FormExample4.razor?highlight=38,42-53,71)]
```

Comment:
Are these basically the same apps just targeting different target frameworks?

Reply:
We have four snippet samples: 3.x and 5.x for each of WASM and Server. That's the only way forward with snippet sample apps that doesn't become a nightmare to maintain. Post 6.0-GA, I'll just duplicate 5.x, update for 6.x with fixes for breaking changes, and add code cross-links as needed. When 3.x drops, 🔪 both 3.x sample apps and 🔪 the cross-links to them. This is the fastest (**_cheapest_** 💲 pro-Hunter budget 💰 thing) that we can do :smile:.